apis/v1alpha1/opentelemetrycollector_webhook.go

```diff
@@ -120,9 +121,19 @@ func (r *OpenTelemetryCollector) validateCRDSpec() error {
 		}
 	}
 
+	// validator port config
+	for _, p := range r.Spec.Ports {
+		nameErrs := validation.IsValidPortName(p.Name)
+		numErrs := validation.IsValidPortNum(int(p.Port))
+		if len(nameErrs) > 0 || len(numErrs) > 0 {
+			return fmt.Errorf("the OpenTelemetry Spec Ports configuration is incorrect, port name '%s' errors: %s, num '%d' errors: %s",
+				p.Name, nameErrs, p.Port, numErrs)
+		}
+	}
+
 	// validate autoscale with horizontal pod autoscaler
 	if r.Spec.MaxReplicas != nil {
-		if *r.Spec.MaxReplicas < int32(1) {
+		if *r.Spec.MaxReplicas <= int32(1) {
```

Comment:
I changed this to match the error string explanation (max replicase should be more than one) but I don't think there's a problem if maxReplicas is 1? I can change the error string instead of changing this expression if others agree.

Reply:
Moved: https://github.com/open-telemetry/opentelemetry-operator/pull/1136

I will remove these changes from this PR.